apps/managedidentity/managedidentity.go

```diff
@@ -204,28 +291,177 @@ func (client Client) getTokenForRequest(req *http.Request) (accesstokens.TokenRe
-// Options: [WithClaims]
-func (client Client) AcquireToken(ctx context.Context, resource string, options ...AcquireTokenOption) (base.AuthResult, error) {
-	o := AcquireTokenOptions{}
+func fileExists(path string) bool {
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
+	}
+	return true
+}
 
-	for _, option := range options {
-		option(&o)
+func createIMDSAuthRequest(ctx context.Context, id ID, resource string) (*http.Request, error) {
+	var msiEndpoint *url.URL
+	msiEndpoint, err := url.Parse(imdsEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't parse %q: %s", imdsEndpoint, err)
+	}
+	msiParameters := msiEndpoint.Query()
+	msiParameters.Set(apiVersionQueryParameterName, imdsAPIVersion)
+	resource = strings.TrimSuffix(resource, "/.default")
+	msiParameters.Set(resourceQueryParameterName, resource)
+
+	switch t := id.(type) {
+	case UserAssignedClientID:
+		msiParameters.Set(miQueryParameterClientId, string(t))
+	case UserAssignedResourceID:
+		msiParameters.Set(miQueryParameterResourceId, string(t))
+	case UserAssignedObjectID:
+		msiParameters.Set(miQueryParameterObjectId, string(t))
+	case systemAssignedValue: // not adding anything
+	default:
+		return nil, fmt.Errorf("unsupported type %T", id)
 	}
-	req, err := createIMDSAuthRequest(ctx, client.miType, resource, o.claims)
+
+	msiEndpoint.RawQuery = msiParameters.Encode()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, msiEndpoint.String(), nil)
 	if err != nil {
-		return base.AuthResult{}, err
+		return nil, fmt.Errorf("error creating http request %s", err)
+	}
+	req.Header.Set(metaHTTPHeaderName, "true")
+	return req, nil
+}
+
+func createAzureArcAuthRequest(ctx context.Context, id ID, resource string) (*http.Request, error) {
+	identityEndpoint, _ := getAndValidateAzureArcEnvVars()
+	var msiEndpoint *url.URL
+
+	if _, ok := id.(systemAssignedValue); !ok {
+		return nil, errors.New("Azure Arc doesn't support user assigned managed identities")
+	}
+
+	msiEndpoint, parseErr := url.Parse(identityEndpoint)
+	if parseErr != nil {
+		return nil, fmt.Errorf("couldn't parse %q: %s", identityEndpoint, parseErr)
 	}
-	tokenResponse, err := client.getTokenForRequest(req)
+
+	msiParameters := msiEndpoint.Query()
+	msiParameters.Set(apiVersionQueryParameterName, azureArcAPIVersion)
+	resource = strings.TrimSuffix(resource, "/.default")
+	msiParameters.Set(resourceQueryParameterName, resource)
+
+	msiEndpoint.RawQuery = msiParameters.Encode()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, msiEndpoint.String(), nil)
 	if err != nil {
-		return base.AuthResult{}, err
+		return nil, fmt.Errorf("error creating http request %s", err)
 	}
-	return base.NewAuthResult(tokenResponse, shared.Account{})
+	req.Header.Set(metaHTTPHeaderName, "true")
+	return req, nil
+}
+
+// GetEnvironmentVariables returns the identity and IMDS endpoints
+func getAndValidateAzureArcEnvVars() (string, bool) {
```

Comment:
I think Neha had a comment that looked like we had 2 methods that were doing the same thing, the validation part, similar to .NET that is used in the GetSource func called ValidateAzureArcEnvironment function
So i extracted that logic into the getAzureArcEndpoint() call, I am going to revert back and have validateAzureArcEnvironment for just the GetSource and then for the AzureRequest we can get the correct endpoint

Reply:
I can try to answer this
1. At this point we are trying to validate that the code is running on an azure arc env. If the env is azure arc the expectation is to find a combination of those 2 env variables. But in some env like linux the env variables are not detected by the process so we check the presence of file and declare that the env is azure arc. 
2. Validate when both are empty because if either is set and not both that means it is some other Azure env where the code is running and not Azure arc. Unfortunately, looking at env variables combination presence is the way to detect if managed identity is available and what Azure env it is.
3. Because while GetSource method only cares about detecting the env, createAzureArcAuthRequest knows that the env is azure arc and only looks for the returned endpoint. 

IMO this can be simplified by not calling this method in createAzureArcAuthRequest and just use the arc endpoint.

Reply:
Removed second function, kept the validateAzureArcEnvironment function
Request just uses the default arc endpoint now since we know its azure arc